packages/teleport/src/lib/term/tty.js

```diff
@@ -156,6 +159,26 @@ class Tty extends EventEmitter {
       this.emit(TermEventEnum.RESIZE, { w, h });
     }
   }
+
+  _processU2FChallenge(payload) {
+    const data = JSON.parse(payload);
+    window.u2f.sign(data.appId, data.challenge, data.u2f_challenges, this._processU2FResponse.bind(this));
+  }
+
+  _processU2FResponse(res) {
+    if (res.errorCode) {
+      let errorMsg = `error code ${res.errorCode}`;
+      // lookup error message...
+      for (var msg in window.u2f.ErrorCodes) {
+        if (window.u2f.ErrorCodes[msg] == res.errorCode) {
+          errorMsg = msg;
+        }
+      }
+      logger.info(`Please check your U2F settings, make sure it is plugged in and you are using the supported browser.\nU2F error: ${errorMsg}`);
```

Comment:
@awly why not writing this message to the terminal?

```
const msg = `Please check your U2F settings, make sure it is plugged in and you are using the supported browser.\nU2F error: ${errorMsg}`;

this.term.write(`\x1b[31m${msg}\x1b[m\r\n`);
```

Reply:
good idea, had to move this code to `terminal.js` to do this.